packages/roosterjs-content-model-markdown/lib/modelToMarkdown/processor/modelProcessor.ts

```diff
@@ -0,0 +1,44 @@
+import { createMarkdownBlockGroup } from '../creators/createMarkdownBlockGroup';
+import { createMarkdownParagraph } from '../creators/createMarkdownParagraph';
+import { createMarkdownTable } from '../creators/createMarkdownTable';
+import type { ListCounter } from '../creators/createMarkdownBlockGroup';
+import type { ContentModelDocument } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function modelProcessor(model: ContentModelDocument) {
+    let markdown = '';
+    const listCounter: ListCounter = {
+        listItemCount: 0,
+        subListItemCount: 0,
+    };
+    for (const block of model.blocks) {
+        const isListItem = block.blockType === 'BlockGroup' && block.blockGroupType === 'ListItem';
+        if (!isListItem && (listCounter.listItemCount > 0 || listCounter.subListItemCount > 0)) {
+            listCounter.listItemCount = 0;
+            listCounter.subListItemCount = 0;
+            markdown += '\n';
```

Comment:
Suggest make the  NewLine char as a parameter, in case user want to use \r\n instead